src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,111 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
 
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double RESIZE_COEFFICIENT = 1.5;
+    private int size;
+    private T[] values;
+
+    public ArrayList() {
+        this.values = (T[]) new Object[DEFAULT_CAPACITY];
+        this.size = 0;
+    }
+
+    public ArrayList(int initialCapacity) {
+        this.values = (T[]) new Object[initialCapacity];
+        this.size = 0;
+    }
+
+    private void resize() {
```

Comment:
let’s move private methods to the bottom of the file, or at least after the place where we call them

Reply:
Outdated code